inc/azure_macro_utils/macro_utils.h

```diff
@@ -322,6 +322,26 @@ __pragma(warning(pop))
 #endif
 
 
+/*the below 2 macros do the same thing: they take a construct in the form of ORIGINAL_NAME(ARG1, ARG2,...) and expand to 
+* SUBST_NAME(ARG1, ARG2,...), so effectively it substitutes one token with another. Here's some example:
+* MU_SUBSTITUTE(SUM(7,3), DIF) instead of doing something with SUM(7,3) will expand to DIFF(7,3). In order for this to work, the user needs to mark "SUM" as a replacebable macro by
```

Comment:
>SUM [](start = 16, length = 3)

Seems like a weird example if I understand the note below "ORIGINAL_NAME must not be defined a macro before or errors will occur". Because the name "SUM" makes me believe there is some macro like:

```
#define SUM(a,b) ((a)+(b))
```

And this lets you change it like a replacement. But the comment below tells me this wouldn't work very well.

Is "ORIGINAL_NAME" meant to be something more generic? Like how we have macros that look like:
DEFINE_SOMETHING(ARG(int, a), ARG(int, b))

And you want to replace ARG with some other macro because ARG on its own is just syntactic sugar? The test case seems to have an example that is more in line with the "ARG" thing than the "SUM"->"DIFF" example

Reply:
Updated to say "PAIR" and updated comments to be better.

---
In reply to: [508001670](https://github.com/Azure/azure-macro-utils-c/pull/49#discussion_r508001670) [](ancestors = 508001670)